oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/query/FulltextIndexTracker.java

```diff
@@ -107,27 +102,27 @@ AsyncIndexInfoService getAsyncIndexInfoService() {
 
     private synchronized void diffAndUpdate(final NodeState root) {
         if (asyncIndexInfoService != null && !asyncIndexInfoService.hasIndexerUpdatedForAnyLane(this.root, root)) {
-            log.trace("No changed detected in async indexer state. Skipping further diff");
+            LOG.trace("No changed detected in async indexer state. Skipping further diff");
             this.root = root;
             return;
         }
 
-        Map<String, IndexNodeManager> original = indices;
-        final Map<String, IndexNodeManager> updates = newHashMap();
+        Map<String, I> original = indices;
+        final Map<String, I> updates = new HashMap<>();
 
-        Set<String> indexPaths = Sets.newHashSet();
+        Set<String> indexPaths = new HashSet<>();
         indexPaths.addAll(original.keySet());
         indexPaths.addAll(badIndexTracker.getIndexPaths());
 
-        List<Editor> editors = newArrayListWithCapacity(indexPaths.size());
+        List<Editor> editors = new ArrayList<>(indexPaths.size());
         for (final String path : indexPaths) {
             editors.add(new SubtreeEditor(new DefaultEditor() {
                 @Override
                 public void leave(NodeState before, NodeState after) {
                     try {
                         if (isStatusChanged(before, after) || isIndexDefinitionChanged(before, after)) {
```

Comment:
I am not 100% sure about this check. This code has been inherited from the Lucene implementation.
When some new content gets indexed, the index definition status is modified with the last update timestamp and the number of updates. Would not be enough to check if the index def has changed? This is quite important because every time we need to update an index definition in the tracker we need to acquire a write lock that could be held by current running searches (see the comment at line 147).
@thomasmueller @nit0906 @tihom88 any feedback on this is appreciated

Reply:
I have to admit I don't understand the reason why this is done for Lucene... I think we need to understand this, otherwise we can't decide if it's needed for Elasticsearch or not...